src/App/Pages/Generator/GeneratorHistoryPageViewModel.cs

```diff
@@ -57,5 +61,29 @@ private async void CopyAsync(GeneratedPasswordHistory ph)
             _platformUtilsService.ShowToast("info", null,
                 string.Format(AppResources.ValueHasBeenCopied, AppResources.Password));
         }
+
+        public void Subscribe()
+        {
+            _broadcasterService?.Subscribe(nameof(GeneratorHistoryPageViewModel), async (message) =>
```

Comment:
The problem is that the `Page` has no idea when the app goes back to Foreground, this is only available on the `AppDelegate -> OnActivated` method; so we need to have the messenger or some sort of pub/sub or event to tell the page that the app is now on foreground again to update the theme of the passwords. That's why I used the `messenger`, I think this is a particular case where the messenger is handy.

For 2., we could subscribe to the event `App.Current.RequestedThemeChanged` on the pages that need to update the theme and apply it there; the problem with that is that this event doesn't actually work well when coming from background to foreground, because the things that needs to applied on foreground on the main thread may be done before the desired page is actually visible, thus the changes may not be applied entirely. So iOS `OnActivated` / Droid `OnResume`, are better places that actually tells that you can do some work on the UI that will be applied correctly to the View.

Maybe there's a better way, but this is the simplest and that works on all cases that I've found.